lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -423,13 +424,13 @@ def is_online_index_pointer(self):
 
 class Requirement(
         _ComputedReqKindsMixin,
-        namedtuple('Requirement', ('fqcn', 'ver', 'src', 'type')),
+        namedtuple('Requirement', ('fqcn', 'ver', 'src', 'type', 'user_defined')),
```

Comment:
Such an extra attribute may negatively impact how `in`-checks work in all the places where we work with the sequences of collections because it's not just a detached attribute, it's an extension of a tuple...